# Ejercicio1 - Utilizar operadores aritméticos

    Operadores aritméticos en Python

Python proporciona operadores aritméticos comunes para que puedas realizar operaciones matemáticas en tu código. Estos incluyen las cuatro operaciones principales de suma, resta, multiplicación y división.

Exploremos cómo podemos crear un programa que pueda calcular la distancia entre dos planetas. Comenzaremos usando dos distancias de planetas: Tierra (149.597.870 km) y Júpiter (778.547.200 km).

In [1]:
tierra = 149597870
jupiter = 778547200

    Realizar la operación

Con los valores obtenidos, es el momento de añadir el código para realizar la operación. Restarás el primer planeta del segundo para determinar la distancia en kilómetros. A continuación, puedes convertir la distancia del kilómetro en millas multiplicándola por 0.621.

In [6]:
distancia_tj = abs(tierra - jupiter)
distancia_tjmillas = distancia_tj * 0.621

print('En kilómetros:')
print(distancia_tj)
print('En millas:')
print (distancia_tjmillas)

En kilómetros:
628949330
En millas:
390577533.93


    Prueba tu proyecto

Con el código creado, ejecuta el notebook para obtener el resultado. Deberías recibir lo siguiente:

628949330
390577534

In [7]:
print('En kilómetros:')
print(distancia_tj)
print('En millas:')
print (distancia_tjmillas)

En kilómetros:
628949330
En millas:
390577533.93


# Ejercicio 2: convierte cadenas en números y usa valores absolutos

    Crear una aplicación para trabajar con números y entrada de usuario

Con frecuencia, deberás convertir los valores de cadena en números para realizar correctamente diferentes operaciones o determinar el valor absoluto de un número.

Para crear nuestra aplicación, queremos leer la distancia del sol para dos planetas, y luego mostrar la distancia entre los planetas. Haremos esto usando input para leer los valores, int para convertir a entero y luego abs para convertir el resultado en su valor absoluto.


    Lee los valores

Usando input, agrega el código para leer la distancia del sol para cada planeta, considerando 2 planetas.


In [1]:
planeta1 = 'tierra'
planeta2 = 'marte'

print("Sistema de medición de distancia entre el sol y distintos planetas")
entrada_planeta1 = input(f'Distancia de la {planeta1} al sol: ')
entrada_planeta2 = input(f'Distancia de {planeta2} al sol: ')

print (f'Distancia de la {planeta1} al sol en kilometros:')
print (entrada_planeta1)

print (f'Distancia de {planeta2} al sol en kilometros:')
print (entrada_planeta2)

Sistema de medición de distancia entre el sol y distintos planetas
Distancia de la tierra al sol en kilometros:
1534153153
Distancia de marte al sol en kilometros:
471615161


    Convertir a número

Debido a que input devuelve valores de cadena, necesitamos convertirlos en números. Para nuestro ejemplo, usaremos int

In [15]:
distancia_planeta1 = int(entrada_planeta1)
distancia_planeta2 = int(entrada_planeta2)

print(distancia_planeta1)
print(distancia_planeta2)

5648489464
1564654564


    Realizar el cálculo y convertir a valor absoluto

Con los valores almacenados como números, ahora puedes agregar el código para realizar el cálculo, restando el primer planeta del segundo. Debido a que el segundo planeta podría ser un número mayor, usarás abs para convertirlo a un valor absoluto. También agregarás el código para mostrar el resultado en millas multiplicando la distancia del kilómetro por 0.621

In [16]:
distancia_planetas = abs(distancia_planeta1 - distancia_planeta2)
distancia_planetasmillas = distancia_planetas * 0.621

print('En kilómetros:')
print(distancia_planetas)
print('En millas:')
print (distancia_planetasmillas)

En kilómetros:
4083834900
En millas:
2536061472.9


    Prueba tu aplicación

Para probar el proyecto, ejecuta tu notebook. En la parte superior de vscode surgirá un cuadro de diálogo para que proporciones las distancias. Puede utilizar los datos de la tabla siguiente:

    Planeta 	Distancia al sol

    Mercurio 	57900000

    Venus 	    108200000

    Tierra 	    149600000

    Marte 	    227900000

    Júpiter 	778600000

    Saturno 	1433500000

    Urano 	    2872500000
    
    Neptuno 	4495100000

In [18]:
planeta1 = 'tierra'
planeta2 = 'marte'

print("Sistema de medición de distancia entre el sol y distintos planetas")
entrada_planeta1 = input(f'Distancia de la {planeta1} al sol: ')
entrada_planeta2 = input(f'Distancia de {planeta2} al sol: ')

print (f'Distancia de la {planeta1} al sol en kilometros')
print (entrada_planeta1)

print (f'Distancia de {planeta2} al sol en kilometros')
print (entrada_planeta2)

distancia_planeta1 = int(entrada_planeta1)
distancia_planeta2 = int(entrada_planeta2)

distancia_planetas = abs(distancia_planeta1 - distancia_planeta2)
distancia_planetasmillas = distancia_planetas * 0.621

print(f'Distancia entre la {planeta1} y {planeta2} en kilómetros:')
print(distancia_planetas)
print(f'Distancia entre la {planeta1} y {planeta2} en millas:')
print (distancia_planetasmillas)

Sistema de medición de distancia entre el sol y distintos planetas
Distancia de la tierra al sol en kilometros
149600000
Distancia de marte al sol en kilometros
227900000
Distancia entre la tierra y marte en kilómetros:
78300000
Distancia entre la tierra y marte en millas:
48624300.0
